receiver/nats.go

```diff
@@ -0,0 +1,163 @@
+/*
+ * skogul, nats receiver/subscriber
+ *
+ * Author(s):
+ *  - Niklas Holmstedt <n.holmstedt@gmail.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301  USA
+ */
+
+package receiver
+
+import (
+	"crypto/tls"
+	"fmt"
+	"github.com/nats-io/nats.go"
+	"github.com/telenornms/skogul"
+	"sync"
+)
+
+var natsLog = skogul.Logger("receiver", "nats")
+
+/*
+Nats.io simple pub/sub receiver with:
+
+- Authentication: Username/Password, TLS
+- Authorization: Username/Password, UserCredentials/JWT
+- Queue: Load balancing for multiple receivers in the same Queue.
+*/
+type Nats struct {
```

Comment:
Same comment regarding Golang Docs as above.

Additionally, I'm not sure what the bullet list refers to? I can't see "Authentication" or "Authorization" referenced in the configuration by those names. Could you update the documentation with which does what?

Reply:
Authorization and Authentication (AuthZ, AuthN) is separated within Nats, so it's more to show that these are implemented. Buy i guess i can document it as a single, since it isn't referenced in the code.